src/slurm_plugin/instance_manager.py

```diff
@@ -968,6 +960,7 @@ def all_or_nothing_node_assignment(
                     print_with_count(nodes_resume_list),
                 )
                 self._update_dict(self.nodes_assigned_to_instances, nodes_resume_mapping)
+                self._reset_failed_nodes(set(nodes_resume_list))
```

Comment:
NIT: It seems that both `best-effort` and a successful `all-or-nothing` handle the successfully launched nodes _roughly_ the same way. Maybe we can have shared behaviour for both cases?

```
def handle_successfully_launched_nodes:
    - Update the node mapping dictionary
    - Reset the failed nodes
```

Reply:
not really, they are different. Let's sync on this